app/quest-boost/[boostId]/page.tsx

```diff
@@ -155,44 +119,39 @@ export default function Page({ params }: BoostQuestPageProps) {
           <p>Reward:</p>
           <div className="flex flex-row gap-2">
             <p className={styles.claim_button_text_highlight}>
-              {boost?.amount} USDC
+              {boost?.amount} {getTokenName()}
             </p>
-            <CDNImage
-              src={"/icons/usdc.svg"}
-              priority
-              width={32}
-              height={32}
-              alt="usdc icon"
-            />
+            <TokenSymbol tokenAddress={boost?.token ?? ""} />
           </div>
           <p>among</p>
           <p className={styles.claim_button_text_highlight}>
             {participants} players
           </p>
         </div>
-        <div>
-          <Button
-            disabled={
-              boost?.claimed ||
-              hexToDecimal(boost?.winner ?? "") !== hexToDecimal(address)
-            }
-            onClick={handleClaimClick}
-          >
-            {(() => {
-              if (boost?.claimed) {
-                return "Claimed ✅";
-              } else if (
-                hexToDecimal(boost?.winner ?? "") === hexToDecimal(address)
-              ) {
-                return "Claim boost reward 🎉 ";
-              } else if (boost && boost?.expiry > Date.now()) {
-                return "Boost has not ended ⌛";
-              } else {
-                return "You’re not selected 🙁";
+        {address ? (
+          <div>
+            <Button
+              disabled={
+                boost &&
+                (boost?.expiry > Date.now() ||
+                  boost?.claimed ||
+                  BoostClaimStatusManager.getBoostClaimStatus(boost.id))
               }
-            })()}
-          </Button>
-        </div>
+              onClick={() => {
+                if (!boost) return;
+                if (hexToDecimal(boost?.winner ?? "") !== hexToDecimal(address))
+                  BoostClaimStatusManager.updateBoostClaimStatus(
+                    boost?.id,
+                    true
+                  );
+
+                router.push(`/quest-boost/claim/${boost?.id}`);
+              }}
```

Comment:
might be worth to use a function for readability of the `onClick`

Reply:
done